src/mlpack/core/util/using.hpp

```diff
@@ -71,6 +73,37 @@ namespace mlpack {
 
 #endif
 
+  namespace internal_compact {
+
+    namespace fill {
+
+      #ifdef MLPACK_HAS_COOT
+        struct fill_none  : public decltype(arma::fill::none),
+                            public decltype(coot::fill::none) { };
+
+        struct fill_zeros : public decltype(arma::fill::zeros),
+                            public decltype(coot::fill::zeros) { };
+
+        struct fill_ones  : public decltype(arma::fill::ones),
+                            public decltype(coot::fill::ones) { };
+
+        struct fill_randu : public decltype(arma::fill::randu),
+                            public decltype(coot::fill::randu) { };
+
+      #else
+        struct fill_none  : public decltype(arma::fill::none) { };
+        struct fill_zeros : public decltype(arma::fill::zeros) { };
+        struct fill_ones  : public decltype(arma::fill::ones) { };
+        struct fill_randu : public decltype(arma::fill::randu) { };
+      #endif
```

Comment:
I think that Conrad's suggestion is a safer way to go here:

```c++
// By default, assume that we are using an Armadillo object.
template<typename MatType>
struct GetFillType
{
  static constexpr decltype(arma::fill::none) none = arma::fill::none;
  static constexpr decltype(arma::fill::zeros) zeros = arma::fill::zeros;
  static constexpr decltype(arma::fill::ones) ones = arma::fill::ones;
  static constexpr decltype(arma::fill::randu) randu = arma::fill::randu;
  static constexpr decltype(arma::fill::randn) randn = arma::fill::randn;
};

#ifdef MLPACK_HAS_COOT
// If the matrix type is a Bandicoot type, use Bandicoot fill objects instead.
template<typename MatType,
         typename = typename std::enable_if<is_coot_type<MatType>::value>::type*>
struct GetFillType
{
  static constexpr decltype(coot::fill::none) none = coot::fill::none;
  static constexpr decltype(coot::fill::zeros) zeros = coot::fill::zeros;
  static constexpr decltype(coot::fill::ones) ones = coot::fill::ones;
  static constexpr decltype(coot::fill::randu) randu = coot::fill::randu;
  static constexpr decltype(coot::fill::randn) randn = coot::fill::randn;
};
#endif
```

I haven't tried compiling that, but the general idea should be adaptable into something that works. :+1: